### PR TITLE
Add Percy snapshot testing.

### PR DESCRIPTION
**Pull request by robdodson**
Adds support for http://percy.io/ snapshot testing. Currently testing specific landing pages. After landing this PR I'd like to do a follow up to let us test our UI library (web components, shortcodes, etc) but that'll take more work.

Changes proposed in this pull request:

- Filter post collections based on test env.
- Add a new GitHub workflow to run Percy tests.
- Set cookie banner to accepted if we're in a test or dev env.

@mfriesenhahn @kaycebasques fyi

### CODE REVIEW

**Comment by robdodson** [comment]
> I presume this just warns us of the screenshot change delta :)

yep, it's quite fancy. I sent you an invite to view the project. It takes shots at a couple different screen sizes and grumps if there are changes. For now i'm just testing Chrome at two sizes. We get 5000 shots a month for free.

File: src/lib/store.js [hunk 1]
```diff
@@ -42,7 +42,9 @@ const initialState = {
 
   // When a user lands on the page, check if they have accepted our
   // cookie policy.
-  userAcceptsCookies: false,
+  // We disable this check in dev and test environments so the cookie banner
+  // doesn't interfere with testing.
+  userAcceptsCookies: config.prod ? false : true,
```
Comment (samthor) [inline src/lib/store.js]: super big nit but I think this might be more readable as

```js
   userAcceptsCookies: !config.prod,
```

Comment (samthor) [reply src/lib/store.js]: and update the comment to have fewer negatives, so e.g. "We automatically accept cookies anywhere but prod so that ..."

**Comment by robdodson** [comment]
@MichaelSolati PTAL when you get a chance

**Pull request merged**